source/develop/release-management/features/virt/high-performance-vm.html.md

```diff
@@ -93,60 +91,67 @@ Displayed in 'Resource Allocation' side-tab of the VM dialog.
 
 #### **Disable the Watchdog device**
 Displayed in 'High Availability' side-tab of the VM dialog.
 
+#### **Enable paravirtualized Random Number Generator PCI (virtio-rng) device** 
+Displayed in 'Random Generator' side-tab of the VM dialog.
+
+This option is automatically set using libvirt defaults ("Period duration"=1000 milliseconds) and can be changed by the user before saving.
+
 ### List of manual configuration settings/warnings for High Performance VM
 
 #### **Set the CPU Pinning topology**
 Displayed in 'Resource Allocation' side-tab of the VM dialog.
 
-In case the CPU Pinning topology is not set upon VM saving,  a recommendation/warning will be displayed in a pop-up and the user will be asked to pin the VM to a host (by selecting the "Start running on: Specific Host" in "Host" side-tab) and verify if VM configuration fits the host configuration (number of cores per socket,  threads per core and number of sockets should match).
+In case the CPU Pinning topology is not set upon VM saving, a recommendation/warning will be displayed in a pop-up and the user will be asked to pin the VM to a host (by selecting the "Start running on: Specific Host" in "Host" side-tab) and verify if VM configuration fits the host configuration (number of cores per socket,  threads per core and number of sockets should match).
 
-Ideally the CPU pinning should be done automatically and not manually by the user, But in first phase we will not support it.  The automatic pinning can be done by pinning to the CPU that is responsible for the IO in the host (and don't present this CPU to the guest) or alternatively pinned to CPU#0 as this is typically the one.  
-
-In addition we will suggest to set the emulatorpin in libvirt xml.
+Ideally the CPU pinning should be done automatically and not manually by the user, But in first phase we will not support it.  
 
 #### **Set the I/O Thread pinning topology**
 This is currently not implemented. 
-Should be added to the 'Resource Allocation' side-tab of the VM dialog. It will be implemented by setting  the iothreadpin in libvirt xml accordingly
+Should be added to the 'Resource Allocation' side-tab of the VM dialog. It will be implemented by setting the iothreadpin in libvirt xml accordingly (requires VDSM changes).
 
-This feature should be implemented the same as the CPU pinning is implemented.
+This feature should be implemented the same as the CPU pinning is implemented. 
+Ideally the the I/O Thread pinning should be done automatically and not manually by the user, But in first phase we will not support it.  The automatic pinning can be done by pinning to the CPU that is responsible for the IO in the host (and don't present this CPU to the guest) or alternatively pinned to CPU#0 as this is typically the one. 
```

Comment:
That can get rather painful with NUMA. I'd prefer the pin to be cpu#0 of the first numa nodes where the guest runs (given that guest is pinned to numa node#2, I believe IO should be done from the node#2). That being said, we should benchmark this.